pyomeca/rototrans.py

```diff
@@ -1,499 +1,273 @@
-            )
+        if data.ndim == 2:
+            data = data[..., np.newaxis]
 
-        # Sanity checks
-        if angles.get_num_frames() != translations.get_num_frames():
-            raise IndexError(
-                "angles and translations must have the same number of frames"
-            )
-        if angles.shape[0] is not len(angle_sequence):
-            raise IndexError("angles and angles_sequence must be the same size")
-        if angles.get_num_frames() == 0:
-            return RotoTrans()
-
-        rt_out = np.repeat(np.eye(4)[:, :, np.newaxis], angles.get_num_frames(), axis=2)
-        try:
-            for i in range(len(angles)):
-                a = angles[i, :, :]
-                matrix_to_prod = np.repeat(
-                    np.eye(4)[:, :, np.newaxis], angles.get_num_frames(), axis=2
-                )
-                if angle_sequence[i] == "x":
-                    # [[1, 0     ,  0     ],
-                    #  [0, cos(a), -sin(a)],
-                    #  [0, sin(a),  cos(a)]]
-                    matrix_to_prod[1, 1, :] = np.cos(a)
-                    matrix_to_prod[1, 2, :] = -np.sin(a)
-                    matrix_to_prod[2, 1, :] = np.sin(a)
-                    matrix_to_prod[2, 2, :] = np.cos(a)
-                elif angle_sequence[i] == "y":
-                    # [[ cos(a), 0, sin(a)],
-                    #  [ 0     , 1, 0     ],
-                    #  [-sin(a), 0, cos(a)]]
-                    matrix_to_prod[0, 0, :] = np.cos(a)
-                    matrix_to_prod[0, 2, :] = np.sin(a)
-                    matrix_to_prod[2, 0, :] = -np.sin(a)
-                    matrix_to_prod[2, 2, :] = np.cos(a)
-                elif angle_sequence[i] == "z":
-                    # [[cos(a), -sin(a), 0],
-                    #  [sin(a),  cos(a), 0],
-                    #  [0     ,  0     , 1]]
-                    matrix_to_prod[0, 0, :] = np.cos(a)
-                    matrix_to_prod[0, 1, :] = -np.sin(a)
-                    matrix_to_prod[1, 0, :] = np.sin(a)
-                    matrix_to_prod[1, 1, :] = np.cos(a)
-                else:
-                    raise ValueError(
-                        "angle_sequence must be a permutation of axes (e.g. "
-                        "xyz"
-                        ", "
-                        "yzx"
-                        ", ...)"
-                    )
-                rt_out = np.einsum("ijk,jlk->ilk", rt_out, matrix_to_prod)
-        except IndexError:
-            raise ValueError(
-                "angle_sequence must be a permutation of axes (e.g. "
-                "xyz"
-                ", "
-                "yzx"
-                ", ...)"
-            )
+        if time is not None:
+            coords["time"] = time
 
-        # Put the translations
-        rt_out[0:3, 3:4, :] = translations[0:3, :, :]
+        # Make sure last line reads [0, 0, 0, 1]
+        data[3, :3, :] = 0
+        data[3, 3, :] = 1
```

Comment:
Should this be verified instead of rewritten?

Reply:
Pyomeca [does the same thing](https://github.com/pyomeca/pyomeca/blob/715daffa5f821dca392b33fe937dcf442e4e832e/pyomeca/rototrans.py#L47-L48). What's weird is when I verify, e.g.:

```python
# Make sure last line reads [0, 0, 0, 1]
zeros = data[3, :3, :]
ones = data[3, 3, :]
if not np.alltrue(zeros == 0) or not np.alltrue(ones == 1):
    some_zeros = np.random.choice(zeros.ravel(), 5)
    some_ones = np.random.choice(ones.ravel(), 5)
    raise ValueError(
        "Last line does not read [0, 0, 0, 1].\n"
        f"Here are some values that should be 0: {some_zeros}\n"
        f"And others that should 1: {some_ones}"
    )
```

I get errors on the `from_transposed_rototrans`, `from_averaged_rototrans` and `from_markers` methods, in both versions of pyomeca (v1 and v2).

Are these two lines necessary in your opinion or is it a bug we have?

Reply:
It may be a "bug" that relies on this feature. Since `from_transposed_rototrans`, `from_averaged_rototrans` and `from_markers` assume the bottom row will be okay, they probably just ignore them instead of setting them properly?

Reply:
Yes that's probably right. So we leave it as this?

Reply:
Yep, we will fix this later

Reply:
I opened #92